### PR TITLE
Updated Python and Django versions in Travis and Tox

### PR DESCRIPTION
**Pull request by awais786**
**Issue:** [BOM-1559](https://openedx.atlassian.net/browse/BOM-1559)

### Changes in Tox and Travis
- Removed `python2.7` and added `python3.8`.
- Removed `django111`, `django20`, and `django21`.
- Updated constraints and made quality fixes.

### CODE REVIEW

**Comment by UsamaSadiq** [comment]
**edx-celeryutils** removed `django-celery` which is causing `import error for djcelery`. 
Trying to fix this issue.

File: requirements/base.txt [hunk 1]
```diff
@@ -8,26 +8,27 @@ amqp==1.4.9               # via kombu
...
+django-model-utils==4.0.0  # via -r requirements/base.in, edx-celeryutils, super-csv
+django==2.2.12            # via -c requirements/constraints.txt, -r requirements/base.in, django-celery, django-crum, django-model-utils, djangorestframework, edx-celeryutils, jsonfield2, super-csv
+djangorestframework==3.11.0  # via super-csv
+edx-celeryutils==0.5.0    # via super-csv
+edx-opaque-keys==2.0.2    # via -r requirements/base.in
 future==0.18.2            # via edx-celeryutils
-idna==2.8                 # via requests
-jsonfield==2.0.2          # via edx-celeryutils
+idna==2.9                 # via requests
+jsonfield2==4.0.0.post0   # via edx-celeryutils
```
Comment (iamsobanjaved) [inline requirements/base.txt:22]: jsonfield2 > 3.0.2 has dropped support for python 3.5. edx-celeryutils has this constraint but still it is pulling latest version. We are good as long as it is not causing trouble?

https://pypi.org/project/jsonfield2/

File: setup.py [hunk 2]
```diff
@@ -82,17 +82,13 @@ def is_requirement(line):
...
-        'Framework :: Django :: 2.1',
         'Framework :: Django :: 2.2',
         'Intended Audience :: Developers',
         'License :: OSI Approved :: GNU Affero General Public License v3 or later (AGPLv3+)',
         'Natural Language :: English',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.5',
+        'Programming Language :: Python :: 3.8',
```
Comment (awais786) [inline setup.py:91]: You dont need to add 3.8.

File: requirements/base.txt (hunk 1 above)
Comment (UsamaSadiq) [reply requirements/base.txt:22]: jsonfield2 version `3.0.3` is the last version to support `python3.5` so pinning `jsonfield2<=3.0.3`.

File: setup.py (hunk 2 above)
Comment (UsamaSadiq) [reply setup.py:91]: Updated setup.py as per your suggestion.

File: requirements/constraints.txt [hunk 3]
```diff
@@ -8,11 +8,17 @@
...
-django-model-utils<4.0.0 ; python_version == "2.7"
\ No newline at end of file
+# zipp>=2.0.0 dropped support for python3.5
+zipp<2.0.0
+
+# jsonfield2>3.0.3 removed support for python3.5
+jsonfield2<=3.0.3
+
+
+mock<4.0.0                         # mock version 4.0.0 drops support for python 3.5
```
Comment (morenol) [inline requirements/constraints.txt:21]: The constraint explanation should be in the line before the actual constraint.

**Pull request merged**